spring-cloud-skipper-server-core/src/main/java/org/springframework/cloud/skipper/server/util/ManifestUtils.java

```diff
@@ -18,18 +18,51 @@
 import java.util.List;
 import java.util.Map;
 import java.util.TreeMap;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
 
 import com.samskivert.mustache.Mustache;
+import org.yaml.snakeyaml.DumperOptions;
+import org.yaml.snakeyaml.Yaml;
+import org.yaml.snakeyaml.nodes.Node;
+import org.yaml.snakeyaml.nodes.Tag;
+import org.yaml.snakeyaml.representer.Representer;
 
 import org.springframework.cloud.skipper.domain.Package;
 import org.springframework.cloud.skipper.domain.Template;
 
 /**
  * Utility functions for manifest file processing.
  * @author Mark Pollack
+ * @author Christian Tzolov
  */
 public class ManifestUtils {
 
+	private final static Yaml yaml = createYaml();
```

Comment:
Yaml is not thread-safe.

Reply:
Ah, important remark. Indeed it is not: https://bitbucket.org/asomov/snakeyaml/wiki/Documentation#markdown-header-threading
Will revert the implementation to create the parser per request.